### DISCUSSION TITLE
How to extract defines from a header?

### DISCUSSION THREAD

**Discussion by stsp** [mesonbuild/meson#12558] [Q&A] [answered]
I tried a stackoverflow for that:
https://stackoverflow.com/questions/77558522/how-to-extract-defines-from-a-header
but seems no one knows an answer.

I'd like to get some defines from a
header and put them into a meson's
variables. I suppose the task can be
achieved by synthesizing `run_command`,
but I'd like meson to assist me with that,
so that I can run the pre-configured
preprocessor, use the pre-set include
paths, and so on.
In a Makefile I have to do:
`VAR = $(shell echo MY_DEFINE | $(CPP) $(CPPFLAGS) -include my_hdr.h - | tail -n 1)`
or something like this. Well, at least
I can use `CPP` and `CPPFLAGS`. But its
far from being ideal.
How can I do the same (or better) with meson?

**Comment by stsp** [comment, marked as answer]
Thanks @dcbaker !
I've got your suggestion to work
and posted an answer on stackoverflow.
```
incdir = include_directories('some_inc_dir')
cpp = meson.get_compiler('c')
VAR = cpp.get_define('MY_DEFINE',
  include_directories: incdir,
  prefix: '#include "my_hdr.h"')
```

**Reply by dcbaker** [reply]
I commented on stack overflow, but the `prefix` argument would save you the need for `-include` here.

**Reply by stsp** [reply]
Thanks, I edited replies to use prefix.
